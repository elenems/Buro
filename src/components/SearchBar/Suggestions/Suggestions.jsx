import React, { Component } from "react";
import { withRouter, Link } from "react-router-dom";

class Suggestions extends Component {
  render() {
    return (
      <div className="suggestions">
        {this.props.items.map(item => {
          return (
            <div key={item.id}>
              <Link to={`/item/${item.id}`}>{item.title}</Link>
            </div>
          );
        })}
      </div>
    );
  }
}

export default withRouter(Suggestions);
