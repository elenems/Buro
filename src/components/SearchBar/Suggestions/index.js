import Suggestions from "./Suggestions.jsx";
export default Suggestions;
