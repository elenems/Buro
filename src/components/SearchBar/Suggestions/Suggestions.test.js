import React from "react";
import { shallow, mount } from "enzyme";
import Suggestions from "../Suggestions";
import { BrowserRouter } from "react-router-dom";

describe("Suggestions", () => {
  it("Renders", () => {
    const wrapper = shallow(<Suggestions />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders suggestions on passed data", () => {
    const props = {
      items: [{ id: 0, title: "First" }, { id: 1, title: "Second" }]
    };
    const wrapper = mount(
      <BrowserRouter>
        <Suggestions {...props} />
      </BrowserRouter>
    );
    expect(wrapper.find("Link").length).toBeGreaterThan(0);
  });
});
