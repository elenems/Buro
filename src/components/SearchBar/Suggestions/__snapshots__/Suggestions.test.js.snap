// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestions Renders 1`] = `
"<ContextConsumer>
  [function]
</ContextConsumer>"
`;
